models/provisioning-jobs.ts

```diff
@@ -0,0 +1,99 @@
+import { HttpMethod } from "../lib/http";
+import { CloudServiceProvider, Network } from "./cloud-service-providers";
+
+export enum ProvisionRequestType {
+  ADD_PORTFOLIO = "ADD_PORTFOLIO",
+  ADD_OPERATORS = "ADD_OPERATORS",
+  ADD_FUNDING_SOURCE = "ADD_FUNDING_SOURCE",
+}
+
+export interface FundingSource {
+  taskOrderNumber: string;
+  clin: string;
+  popStartDate: string;
+  popEndDate: string;
+}
+
+export interface NewPortfolioPayload {
+  name: string;
+  fundingSources: Array<FundingSource>;
+  operators: Array<string>;
+}
+
+export interface FundingSourcePayload {
+  fundingSources: Array<FundingSource>;
+}
+
+export interface OperatorPayload {
+  operators: Array<string>;
+}
+
+export interface ProvisionRequest {
+  jobId: string;
+  userId: string;
+  portfolioId: string;
+  operationType: ProvisionRequestType;
+  // could not use the CloudServiceProviders static properties so used strings
+  targetCsp: "CSP_A" | "CSP_B" | "CSP_C" | "CSP_D";
```

Comment:
You could use `CSP_A.name` instead though right?

Reply:
No typescript would complain. When I try to use the `CSP_A.name` I get the following errors:
- `'CloudServiceProvider' only refers to a type, but is being used as a namespace here.`
- `Property 'targetCsp' of exported interface has or is using private name 'CloudServiceProvider'`

Then in the `start-provisioning-job.ts` I get another error when constructing the `endpoint` (i.g, `${CloudServiceProvider[targetCsp].uri}/portfolios`)
- `Element implicitly has an 'any' type because expression of type 'CloudServiceProvider.CSP_A.name' can't be used to index type 'typeof CloudServiceProvider'.`

This was the option that prevented errors. I'm open to other suggestions though if there is a better way.

Reply:
Is that all because `name` isn't `public`? If it's any more complicated then that no worries.

Reply:
I did try to make `name`  `public`, but that did not solve it either unfortunately.